pkg/apis/core/validation/validation.go

```diff
@@ -3010,6 +3010,32 @@ func validateInitContainers(containers []core.Container, otherContainers []core.
 	return allErrs
 }
 
+func validateHostUsers(spec *core.PodSpec, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
+		if spec.SecurityContext != nil && spec.SecurityContext.HostUsers != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("hostUsers"), "field not supported. Enable the feature gate UserNamespacesSupport"))
+		}
+		return allErrs
+	}
+
+	// Only check security context if hostUsers is false (userns is enabled).
```

Comment:
New new suggestion

```suggestion
	// Only make the following checks if hostUsers is false (otherwise, the container uses the
	// same userns as the host, and so there isn't anything to check).
```

I do think that'll be more clear. The current text could be interpreted as “check if hostUsers is false; that bool comparison is the only comparison to make” by someone who is not sure what the “only” applies to.

Unambiguous comments help new folks get up to speed with the code.